plugins/notifications-backend/src/service/router.ts

```diff
@@ -102,7 +102,7 @@ export async function createRouter(
     const entities = await catalogClient.getEntitiesByRefs(
       {
         entityRefs: refs,
-        fields: ['kind', 'metadata.name', 'metadata.namespace'],
+        fields: ['kind', 'metadata.name', 'metadata.namespace', 'relations', 'spec.children'],
```

Comment:
This caught my eye. There should be no reason whatsoever to get `spec.children`; code should be using `relations` in the first place instead of reading the raw input data from the spec that then generated the relations.

[The code here](https://github.com/backstage/backstage/blob/2ca9dc00b2a1d26cbf8f4700463bdeeb962fec55/plugins/notifications-backend/src/service/router.ts#L125) needs to be changed to use the `type: parentOf` relations instead. Ideally it should also dive more than one level deep, but that's unrelated and could be deferred.

@marleypowell are you up for making that change and also removing the `spec.children` from these two lists again?

Ping @mareklibra @drodil

Reply:
Thanks for spotting this. Indeed it needs to use the relations. Something like this perhaps:

```ts
const childGroupRefs = entity.relations
          .filter(
            relation =>
              relation.type === RELATION_PARENT_OF && relation.targetRef,
          )
          .map(r => r.targetRef);
const childGroups = await catalogClient.getEntitiesByRefs(
          {
            entityRefs: childGroupRefs,
            fields: ['kind', 'metadata.name', 'metadata.namespace', 'relations'],
          },
          { token },
        );
```

Reply:
Additionally, `'relations'` should be probably added here: https://github.com/backstage/backstage/blob/2ca9dc00b2a1d26cbf8f4700463bdeeb962fec55/plugins/notifications-backend/src/service/router.ts#L105

Reply:
Updated with suggested changes

Reply:
Couple of other things as well, already got another PR started: https://github.com/backstage/backstage/pull/24120 😆

Reply:
@drodil if you think this pr as it stands looks good, better to merge it as a nice outside contribution :) feel free to press the button, so it gets into the release today